client/fuzz_test.go

```diff
@@ -0,0 +1,60 @@
+package client_test
+
+import (
+	"context"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+type fuzzSuite struct {
+	IntegrationTestSuite
+}
+
+func (fz *fuzzSuite) FuzzQueryBalance(f *testing.F) {
```

Comment:
we should try to adopt a way for all the queries being fuzzed, one offs make it look weird. What do you think?

Reply:
Perhaps we can add these later on as a task for the quarter. I put this PR up because one day I was hunting for vulnerabilities and figuring out how to take down a node using QueryBalance so that we could fix it. I wrote this test
```go
package client_test

import (
        "context"
        "testing"

        "google.golang.org/grpc"

        "github.com/cosmos/cosmos-sdk/x/bank/types"
)

func TestNodeCrashPanics(t *testing.T) {
        conn, err := grpc.Dial("localhost:9090", grpc.WithInsecure(), grpc.WithBlock())
        if err != nil {
                panic(err)
        }
        defer conn.Close()

        tc := types.NewQueryClient(conn)
        _, err = tc.Balance(
                context.Background(),
                &types.QueryBalanceRequest{
                        Address: "cosmos1wrq8cagsama0xwf2vmlzgrkyynfsxuyhturvyz",
                        Denom:   "00000",
                },
        )
        if err != nil {
                panic(err)
        }
}
``` 

and started going to town so the fuzzer helped out with generating inputs, so it wasn't really planned but I produced content that we could use that.